plugin/supervisor.go

```diff
@@ -17,6 +18,7 @@ import (
 )
 
 type supervisor struct {
+	mut         sync.RWMutex
```

Comment:
nit: could we call this `lock` instead of `mut`?

Reply:
Idiomatic Go suggests that mutexes end with with the text "mut" mentioning the field they protect. And in this case, since it is protecting the entire struct, I have mentioned `mut`. 2/5.

Reply:
Interesting -- the examples I can find via https://github.com/golang/go/search?p=5&q=sync%2Fmutex&unscoped_q=sync%2Fmutex (if that's a good way to search for idiomactic code) use `m`, `mu` or `mutex`, but with no fixed pattern. Perhaps more significantly, though, 7/9 of our usages of `sync.Mutex` or `sync.RWMutex` in this repository use `*Lock` (one uses `mux` and one uses `mutex`).

I do prefer idiomatic Go, but I'm not seeing a strict pattern there. I'd be inclined to stick with `lock` myself, but not going to hold up the review on this though. `mut`, however, seems entirely unusual.

Reply:
Sure, we can go with lock. No worries.

Reply:
Sure, we can go with lock. No worries.